controllers/storagecluster/cephtoolbox.go

```diff
@@ -117,21 +119,36 @@ func removeDuplicateTolerations(tolerations []corev1.Toleration) []corev1.Tolera
 	return list
 }
 
-func getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
+func (r *StorageClusterReconciler) getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
 	err := validateMultusSelectors(sc.Spec.Network.Selectors)
 	if err != nil {
 		return "", err
 	}
 
-	multusNetNs, err := util.GetOperatorNamespace()
-	if err != nil {
-		return "", err
+	if sc.Spec.Network.Selectors["public"] == "" {
+		return "", nil
 	}
 
-	if sc.Spec.Network.Selectors["public"] != "" {
-		multusNetName := sc.Spec.Network.Selectors["public"]
-		return fmt.Sprintf("%s/%s", multusNetNs, multusNetName), nil
+	multusNetName := sc.Spec.Network.Selectors["public"]
+	multusNetNamespacedName := strings.Split(multusNetName, "/")
+	nadNS, nadName := "", ""
+	if len(multusNetNamespacedName) == 2 {
+		nadNS, nadName = multusNetNamespacedName[0], multusNetNamespacedName[1]
+	} else if len(multusNetNamespacedName) < 2 {
+		nadNS, err = util.GetOperatorNamespace()
+		if err != nil {
+			return "", err
+		}
+		nadName = multusNetName
+	} else {
+		return "", fmt.Errorf("public selector value: %s, is not as per specification", multusNetName)
```

Comment:
I think we need to enhance the error msg.

Reply:
How about this?
`return "", fmt.Errorf("public network selector for multus: %s, is invalid", multusNetName)`